tmt/steps/execute/__init__.py

```diff
@@ -55,32 +58,101 @@
 # Metadata file with details about the current test
 TEST_METADATA_FILENAME = 'metadata.yaml'
 
-# Scripts source directory
+#: Scripts source directory
 SCRIPTS_SRC_DIR = tmt.utils.resource_files('steps/execute/scripts')
 
+#: The default scripts destination directory
+SCRIPTS_DEST_DIR = Path("/var/tmp/tmt/bin")  # noqa: S108 insecure usage of temporary dir
+
 
 @dataclass
 class Script:
-    """ Represents a script provided by the internal executor """
+    """
+    Represents a script provided by the internal executor.
+
+    Must be used as a context manager. The context manager returns
+    the source file path.
+
+    The source file name matches the name of the file specified via
+    the ``path`` attribute and must be located in the directory specified
+    via :py:data:`SCRIPTS_SRC_DIR` variable.
+    """
 
     path: Path
     aliases: list[Path]
     related_variables: list[str]
 
+    def __enter__(self) -> Path:
+        return SCRIPTS_SRC_DIR / self.path.name
```

Comment:
This should probably return `effective_scripts_dest_dir()` instead of the default value.

Reply:
Nevermind, I mistook it for `DEST` variable.

Reply:
hmm, how did I miss that

Reply:
```suggestion
        return  effective_scripts_dest_dir() / self.path.name
```

Reply:
oh, I was not blind, ok :) did nothing